mesonbuild/compilers/compilers.py

```diff
@@ -386,7 +386,7 @@ def __init__(self, compiled, returncode=999, stdout='UNDEFINED', stderr='UNDEFIN
         self.stdout = stdout
         self.stderr = stderr
 
-class CompilerArgs(T.MutableSequence[str]):
+class CompilerArgs(collections.abc.MutableSequence):
```

Comment:
This changes the meaning from mypy's point of veiw, you've actually taken away the type annotation, because now mypy doesn't see this as a container of strings, it sees it as a container of Any.

I guess in the short term this is fine, in the long run I'd rather see something like @mensinda's annotation remover. There is a way to publish multiple wheels of a python package for different versions of python as well, so we could in theory publish a 3.5 version with the annotations stripped and a 3.6+ version with the annotations.

Reply:
Maybe we can use a variable that changes based on the python version?

Reply:
That might be possible. On the other hand, python 3.5 is EOL, so I think I'd prefer doing some kind of post processing if we really need to keep supporting it indefinitely